Add quests actions to Habitica

tr4nt0r opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

Adds actions for quest handling in Habitica

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/129650
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced new quest management actions in the Habitica integration:
		- Accept or reject pending quest invitations.
		- Leave an ongoing quest.
		- Abort or start quests as a leader.
		- Cancel quests that have not yet started.
	- Added an automation example for auto-accepting quest invitations, enhancing user experience by streamlining quest participation notifications.

These enhancements provide users with improved control over their quest participation directly from Home Assistant.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tr4nt0r committed:
Add quests actions to Habitica

c0ffeeca7 commented on source/_integrations/habitica.markdown:
```diff
@@ -130,6 +130,30 @@ Use a skill or spell from your Habitica character on a specific task to affect i
+
+### Action `habitica.cancel_quest` 🔒
+
+Cancel a quest that has not yet startet. All accepted and pending invitations will be canceled and the quest roll returned to the owner's inventory. Only quest leader or group leader can perform this action.
```
```suggestion
Cancel a quest that has not yet started. All accepted and pending invitations will be canceled, and the quest roll returned to the owner's inventory. Only the quest leader or group leader can perform this action.
```

c0ffeeca7 commented on source/_integrations/habitica.markdown:
```diff
@@ -130,6 +130,30 @@ Use a skill or spell from your Habitica character on a specific task to affect i
+
+### Action `habitica.start_quest` 🔒
+
+Begin the quest immediately, bypassing any pending invitations that haven't been accepted or rejected. Only quest leader or group leader can perform this action.
```
```suggestion
Begin the quest immediately, bypassing any pending invitations that haven't been accepted or rejected. Only the quest leader or group leader can perform this action.
```

c0ffeeca7 commented on source/_integrations/habitica.markdown:
```diff
@@ -130,6 +130,30 @@ Use a skill or spell from your Habitica character on a specific task to affect i
+
+### Action `habitica.abort_quest` 🔒
+
+Terminate your party's ongoing quest. All progress will be lost and the quest roll returned to the owner's inventory. Only quest leader or group leader can perform this action.
```
```suggestion
Terminate your party's ongoing quest. All progress will be lost, and the quest roll returned to the owner's inventory. Only the quest leader or group leader can perform this action.
```

c0ffeeca7 committed:
tiny tweaks

c0ffeeca7 reviewed (dismissed):
Thank you, @tr4nt0r :+1:

tr4nt0r committed:
Add example automation

tr4nt0r committed:
formating

tr4nt0r committed:
add parameters

tr4nt0r commented:
Added an example automation/blueprint and incorporated some of the coderabbit suggestions. Although not sure about the positioning of the Note box below the actions section. Maybe at the beginning of the actions section or were the restricted actions begin are better positions?

c0ffeeca7 commented:
> Added an example automation/blueprint and incorporated some of the coderabbit suggestions. Although not sure about the positioning of the Note box below the actions section. Maybe at the beginning of the actions section or were the restricted actions begin are better positions?

Nice! :tada: 

Note: Yes, that one is tricky, because people probably don't read top to bottom. 
I agree with you, I would put it right under the first occurrence

tr4nt0r committed:
Move note box

c0ffeeca7 reviewed (approved):
Thank you, @tr4nt0r :+1:
